### ISSUE TITLE
[fit] Improvements

### ISSUE THREAD

**Issue by PiRK** [silx-kit/silx#257]
More improvements for next release (#249 has milestone 0.3)
- [x] add smoothing to strip
- [x] documentation for creating custom FitConfig dialog widgets
- [x] write tutorial and example
- [x] handle background functions the same way as regular fit functions / theories (handle adding custom functions, custom configuration dialog…)
- [x] test estimation, instead of only fit result (position parameters)
- [x] when clicking again on the FIT icon  in PlotWindow, don't open new FitWidget but use existing one (take focus ?)
- [x] adjust content margins around parameters table to reduce wasted space (important on small screens)
- [ ] FitWidget window must not stay on top when user selects the PlotWindow
- [x] Add better tooltips to FitConfig (currently there is a bug: same tooltip applied to all checkboxes on the constraint tab) 
- [x] remove periodic gaussians (too specific)
- [x] rename "Split gaussian" to "Asymetric gaussian"? Could be a problem for PyMca compatibility.
- [x] add buffer to y data (+- search_fwhm) prior to peak search
- [x] autoscaling
- [x] weight in fit

Some of the bigger items will  eventually be moved to a dedicated issue.

**Comment by vasole** [comment]
- Adjust content margins and spacing to reduce "wasted" space
- On my machine, the table is always on top of the plot, even if the plot is a top level window and takes the focus.

**Comment by PiRK** [comment]
About multiple clicks on the Fit button, should we always re-open the same fit window (https://github.com/silx-kit/silx/commit/5c5b696a99f258c798a87ed3a516cc4c010ef288)  or should we allow having one fit widget per curve (https://github.com/silx-kit/silx/commit/5c5b696a99f258c798a87ed3a516cc4c010ef288)?

Is there sometimes a need to fit two different curves side-by-side?

**Comment by vasole** [comment]
The same window. Nobody has ever requested me the side-by-side fit in more than 16 years. Let's say the fit window belongs to the plot window and not to the curve.

**Comment by vasole** [comment]
- To name "runfit" the method doing the actual fit and "fit" the method doing an estimation of parameters and a fit is unfortunate.

- The documentation is wrong. One does not remove a "Constant" or "Linear" background. Those are part of the fitting function. I'm going to take care of this one.

**Comment by PiRK** [comment]
I'm still not sure why the fit window stays on top on windows. I need to investigate the default windowFlags(), now that I can reproduce the issue.

**Comment by PiRK** [comment]
Small progress: i was able to replicate the issue with a simple program.

```
from silx.gui import qt

app = qt.QApplication([])

mw1 = qt.QMainWindow()
mw1.show()
mw1.setWindowTitle("1")

mw2 = qt.QMainWindow(mw1)
mw2.show()
mw2.setWindowTitle("2")

app.exec_()
```

On linux (PyQt4), mw2 can be underneath or on top of mw1. On windows, however, it stays on top (tested with PyQt5).

**Comment by vasole** [comment]
Try not to give a parent to mw2.

**Comment by vasole** [comment]
closed by #519
